File: Gombang/[LeetCode75] - [Two_Pointers]/11_Container_With_Most_Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;

        int maxAreaValue = 0;
        while (left < right)
        {
            int width = right - left;
            int minHeight = min(height[left], height[right]);
            int area = width * minHeight;

            maxAreaValue = max(maxAreaValue, area);

            if (height[left] <= height[right])
                left++;
            else
                right--;
        }

        return maxAreaValue;
    }
};

File: Seori/LeetCode 75/Two Pointers/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        # [1] 투 포인터를 이용하여 높이가 낮은 쪽의 포인터를 이동시킨다.
        left, right = 0, len(height) - 1
        answer = min(height[right], height[left]) * (right - left)
        while left < right:
            if height[left] < height[right]:
                left += 1
            else: right -= 1
            answer = max(answer, (min(height[right], height[left]) * (right - left)))

        return answer